src/net_processing.cpp

```diff
@@ -1430,11 +1468,17 @@ bool static AlreadyHave(const CInv& inv, const CTxMemPool& mempool) EXCLUSIVE_LO
     return true;
 }
 
-void RelayTransaction(const uint256& txid, const CConnman& connman)
+void RelayTransaction(const uint256& txid, const uint256& wtxid, const CConnman& connman)
 {
-    connman.ForEachNode([&txid](CNode* pnode)
+    connman.ForEachNode([&txid, &wtxid](CNode* pnode)
     {
-        pnode->PushTxInventory(txid);
+        AssertLockHeld(cs_main);
+        CNodeState &state = *State(pnode->GetId());
```

Comment:
We shouldn't be indiscriminately taking the address of a return value which may be `nullptr`. I know the same pattern exists in a few other places, but really we should check for existence here before dereferencing the pointer.

Reply:
addressed in https://github.com/bitcoin/bitcoin/pull/19879/commits/fc66d0a65cdc52a3b259effe0c29b5eafb1b5ff5, #19879